src/Models/BaseElement.php

```diff
@@ -694,14 +699,37 @@ public function getSummary()
         return '';
     }
 
+
+    /**
+     * @return array
+     */
+    public function getSummaryData()
+    {
+        $summaryData = [];
+
+        $this->extend('updateSummaryData', $summaryData);
```

Comment:
The problem with adding an extension point here is that the extension runs before overloaded methods do.

Instead I think we need to do something like this:

```php
public function getSummaryData()
{
    $summaryData = $this->provideSummaryData();
    $this->extend('updateSummaryData', $summaryData);
    return $summaryData;
}

/**
 * Overload this method in order to customise the data fields for React elements, this data is serialised
 * and returned via the GraphQL API
 *
 * @return array
 */
protected function provideSummaryData()
{
    return [];
}
```

Reply:
The original intention was to allow people to "extend" in defaults. I guess we could to the provider and all that but I'd rather just remove the extension point at this point if the API is likely to change with upcoming work (and I think it almost certainly will)